test/connection_test.exs

```diff
@@ -365,6 +365,11 @@ defmodule ConnectionTest do
              assert_receive {:EXIT, ^pid, {:shutdown, _}}
              Logger.flush()
            end) =~ ~r"error.*GenServer.*\(stop\) shutdown: #Function.*State: 1"sm
```

Comment:
@josevalim do we still need to do something like this, or is this related to `proc_lib`/`gen`?

Reply:
The error message can change, yes.

Reply:
Right, but in particular I mean, do we still need to do manual logging here? I thought that by implementing `format_status` to return only the `Connection` module state (rather than the gen_statem state), we'd be fine?

Reply:
I would perhaps drop the additional logging because both `format_status` and the logging API changed on OTP since this project was released.